File: Source/trunk/ChromeExt/src/js/Card.js

var Card = function (params) {
	this.id = null;
	this.name = null;
	this.description = null;
	this.owner = 0;
	this.columnId = 0;
	this.boardId = 0;
	this.priority = 0;
	this.checklistCount = 0;
	this.commentCount = 0;
	this.runningTimeLog = 0;
	this.currentEstimated = 0;
	this.attachedCount = 0;
	this.style = 'default';
	this.created = new Date();
	this.updated = null;
	this.status = 1;

	if (IS_INDEX) {
		this.wrapper = null;
		this.header = null;
		this.content = null;
		this.footer = null;
		this.avatar = null;
		this.addAvatarBtn = null;
		this.timerCount = null;
		this.timerPlayIcon = null;
		this.timerStopIcon = null;
		this.timerText = null;
		this.zoomOut = null;
		this.zoomOutIcon = null;
		this.textContent = null;
		this.status = null;
		this.checklistIcon = null;
		this.checklist = null;
		this.commentIcon = null;
		this.comment = null;
		this.attachedIcon = null;
		this.attached = null;
		this.estimated = null;
		this.zoomIn = null;
		this.columnContain = null;
	}

	if (typeof params === 'string') {
		Card.getById(params, function (card) {
			this.id = card.id;
			this.name = card.name;
			this.description = card.description;
			this.owner = card.owner;
			this.columnId = card.columnId;
			this.boardId = card.boardId;
			this.priority = card.priority;
			this.checklistCount = card.checklistCount;
			this.commentCount = card.commentCount;
			this.runningTimeLog = card.runningTimeLog;
			this.currentEstimated = card.currentEstimated;
			this.attachedCount = card.attachedCount;
			this.style = card.style;
			this.created = card.created;
			this.updated = card.updated;
			this.status = card.status;
		});
	} else if (typeof params === 'object') {
		this.id = params.id;
		this.name = params.name;
		this.description = params.description;
		this.owner = params.owner;
		this.columnId = params.columnId;
		this.boardId = params.boardId;
		this.priority = params.priority;
		this.checklistCount = params.checklistCount;
		this.commentCount = params.commentCount;
		this.runningTimeLog = params.runningTimeLog;
		this.currentEstimated = params.currentEstimated;
		this.attachedCount = params.attachedCount;
		this.style = params.style;
		this.created = params.created;
		this.updated = params.updated;
		this.status = params.status;
	} else {
		this.id = generateId('task');
	}

}

File: Source/trunk/SqlServerStorage/Models/Card.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Sioux.TaskManagement.Models
{
	[Table("Cards")]
	public class Card
	{
		[Key]
		public Guid Id { get; set; }

		[Required]
		public string Name { get; set; }

		public string Description { get; set; }

		[Required]
		public int Owner { get; set; }

		[Required]
		public Guid ColumnId { get; set; }

		[Required]
		public Guid BoardId { get; set; }

		[Required]
		public int Priority { get; set; }

		[Required]
		public string Style { get; set; }

		[Required]
		public int CheckListCount { get; set; }

		[Required]
		public int CommentCount { get; set; }

		[Required]
		public int RunningTimeLog { get; set; }

		[Required]
		public int CurrentEstimated { get; set; }

		[Required]
		public int AttachedCount { get; set; }

		[Required]
		public DateTime Created { get; set; }

		[Required]
		public DateTime Updated { get; set; }

		[Required]
		public Status Status { get; set; }

		//[ForeignKey("ColumnId")]
		//public virtual Column Column { get; set; }

		public static implicit operator CardInfo(Card card)
		{
			return new CardInfo()
			{
				Id = card.Id,
				Name = card.Name,
				Description = card.Description,
				Owner = card.Owner,
				BoardId = card.BoardId,
				ColumnId = card.ColumnId,
				Priority = card.Priority,
				Style = card.Style,
				CheckListCount = card.CheckListCount,
				CommentCount = card.CommentCount,
				RunningTimeLog = card.RunningTimeLog,
				CurrentEstimated = card.RunningTimeLog,
				AttachedCount = card.AttachedCount,
				Created = card.Created,
				Updated = card.Updated,
				Status = card.Status
			};
		}
	}
}
